dangerzone/document.py

```diff
@@ -65,10 +79,15 @@ def input_filename(self, filename: str) -> None:
         self.validate_input_filename(filename)
         self._input_filename = filename
 
+        # set the default output filename as soon as we know the input filename
+        self.output_filename = (
+            f"{os.path.splitext(self.input_filename)[0]}{SAFE_EXTENSION}"
+        )
```

Comment:
This scares me a bit because if we do `self.input_filename = ...` after `self.output_filename = ...`, we will overwrite the output filename. It's best to have this as part of `def output_filename()` instead:

```python
      @property
      def output_filename(self) -> str:
          if self._output_filename is None:
              if self._input_filename is not None:
                  # Basically repurpose set_default_output_filename()
                  return self._default_output_filename()
              else:
                  raise errors.NotSetOutputFilenameException()
          else:
              return self._output_filename
```